src/lp/fixed_variable.jl

```diff
@@ -1,18 +1,62 @@
+"""
+    FixedVariableRule
+
+Eliminate all fixed variables.
+
+A variable ``x_{j}`` is fixed if its current lower and upper bounds are equal.
+
+## Presolve
+
+A row
+```math
+l_{i} \\leq a_{i, j}x_{j} + \\sum_{k \\neq j} a_{i, k} x_{k} \\leq u_{i}
+```
+is transformed into
+```math
+l_{i} - a_{i, j} \\bar{x}_{j} \\leq \\sum_{k \\neq j} a_{i, k} x_{k} \\leq u_{i} - a_{i, j} \\bar{x}_{j}
+```
+
+## Postsolve
+
+
+
+For dual variables, first compute
+```math
+z_{j} = c_{j} - \\sum_{i} a_{i, j} (y_{i}^{l} - y_{i}^{u}),
+```
+then recover ``z_{j}^{l} = z_{j}^{+}`` and ``z_{j}^{u} = z_{j}^{-}``.
+
+
+## Misc
+
+* This is a primal reduction.
+* If ``x_{j}`` is integer and fixed to a fractional value, then the problem in infeasible.
+"""
+struct FixedVariableRule <: AbstractPresolveRule end
+
 struct FixedVariable{T} <: PresolveTransformation{T}
     j::Int  # variable index
     x::T  # primal value
     c::T  # current objective coeff
     col::Col{T}  # current column
 end
 
-function remove_fixed_variable!(ps::PresolveData, j::Int)
+function apply!(::FixedVariableRule, ps::PresolveData)
```

Comment:
Style nit: I thought it was standard practice for the ``!`` to denote that the first argument is being mutated. However, I couldn't find that in the Julia style guide right now so maybe not...

Reply:
That was my understanding too, though there are some counter-examples, e.g. [`Base.read!`](https://docs.julialang.org/en/v1/base/io-network/#Base.read!) and [`Base.map!`](https://docs.julialang.org/en/v1/base/collections/#Base.map!).

Here I thought it would make more sense (also when reading the code) to have something like
```julia
apply!(presolve_rule, presolve, args...)
```
which is similar to `map!`, and where the first argument is just used for dispatch.

Reply:
What would the args be? If you want to configure how the rule operates, those could hyperparameters could be fields in the presolve rule type.

Reply:
At this point, the only additional arguments would be numerical tolerances.
For those, I think it makes more sense to have them stored centrally rather than individually for each rule.

In this approach, the main loop would call
```julia
apply!(FixedVariableRule(), ps, ϵ)
```
which, internally, then calls
```julia
remove_fixed_variable!(ps, j, ϵ)
```

If tolerances were stored within the `FixedVariableRule` struct, then these would have to be initialized first
```julia
# Initialize tolerances
rules = [FixedVariableRule(ϵ = 1e-8)]
...
for rule in rules
    apply!(rule, ps)
end
```
(the exact syntax and numerical values are for illustration only)

Reply:
I would argue that tolerances should be encapsulated in a presolve configuration type that can potentially be shared across different Rules. So maybe the signature could just be ``apply!(presolve_rule, presolve, config)``.

Reply:
for reference in the style guide:
https://docs.julialang.org/en/v1/manual/style-guide/#Write-functions-with-argument-ordering-similar-to-Julia-Base

Reply:
I've settled for the convention `apply!(ps, rule, config)`.

I had to change the `_return_if_inferred` macro to accommodate that change.